[Deprecation Reporting] Stub implementation

brentfulgham opened the pull request:
#### c9b4dccdaf956edcacbf76cd8b927420ae0129f7
<pre>
[Deprecation Reporting] Stub implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=245245">https://bugs.webkit.org/show_bug.cgi?id=245245</a>
&lt;rdar://problem/99988574&gt;

Reviewed by NOBODY (OOPS!).

Import the Web Platform Tests for the Deprecation Reporting API, create a new Experimental Feature flag to represent the
feature, and provide a stub implementation of the Deprecation Report Body.

Further work will be done to issue reports once the specification is clarify under what circumstances reports should
be generated.

* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/deprecation-reporting/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/deprecation-reporting/idlharness.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/deprecation-reporting/idlharness.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/deprecation-reporting/idlharness.any.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/deprecation-reporting/idlharness.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/deprecation-reporting/idlharness.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/deprecation-reporting/w3c-import.log: Added.
* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/reporting/DeprecationReportBody.cpp: Added.
(WebCore::DeprecationReportBody::testReportType):
(WebCore::DeprecationReportBody::DeprecationReportBody):
(WebCore::DeprecationReportBody::create):
(WebCore::DeprecationReportBody::type const):
(WebCore::DeprecationReportBody::id const):
(WebCore::DeprecationReportBody::anticipatedRemoval const):
(WebCore::DeprecationReportBody::message const):
(WebCore::DeprecationReportBody::sourceFile const):
(WebCore::DeprecationReportBody::lineNumber const):
(WebCore::DeprecationReportBody::columnNumber const):
(WebCore::DeprecationReportBody::createReportFormDataForViolation):
* Source/WebCore/Modules/reporting/DeprecationReportBody.h: Copied from Source/WebCore/Modules/reporting/TestReportBody.h.
(WebCore::DeprecationReportBody::encode const):
(WebCore::DeprecationReportBody::decode):
(isType):
* Source/WebCore/Modules/reporting/DeprecationReportBody.idl: Copied from Source/WebCore/Modules/reporting/ViolationReportType.h.
* Source/WebCore/Modules/reporting/TestReportBody.h:
* Source/WebCore/Modules/reporting/ViolationReportType.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSReportBodyCustom.cpp:
(WebCore::toJSNewlyCreated): Add support for the new DeprecationReportBody.
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/loader/PingLoader.cpp:
(WebCore::PingLoader::sendViolationReport): Add support for the new DeprecationReportBody.
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;RefPtr&lt;WebCore::ReportBody&gt;&gt;::encode): Add support for the new DeprecationReportBody.
(IPC::ArgumentCoder&lt;RefPtr&lt;WebCore::ReportBody&gt;&gt;::decode): Ditto.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb948b45920726b60586ab893a85df28faafeb6d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89353 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33911 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/20121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/98687 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/154991 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/93361 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32416 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/27928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81713 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/93106 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95000 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/25754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/76284 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/25692 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/80633 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/20121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/68674 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30183 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/20121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29913 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/20121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3177 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33361 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/76284 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32069 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/20121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM assuming the bots are happy.

brentfulgham committed:
Implement no-quirks mode for media/plugin/ua-inline documents
https://bugs.webkit.org/show_bug.cgi?id=228832
<rdar://problem/81851619>

Reviewed by Chris Dumez.

Update Chris Dumez's original patch for 2022, taking into account Bug 244950,
which addressed Text documents.

* LayoutTests/media/modern-media-controls/media-documents/media-document-audio-ios-sizing.html: Update for non-quirks behavior.
* LayoutTests/media/modern-media-controls/media-documents/media-document-audio-ios-sizing-expected.txt: Ditto.
* LayoutTests/platform/ipad/media/modern-media-controls/media-documents/media-document-audio-ios-sizing.html: Ditto.
* LayoutTests/platform/ipad/media/modern-media-controls/media-documents/media-document-audio-ios-sizing-expected.txt: Ditto.
* LayoutTests/platform/mac/editing/pasteboard/copy-standalone-image-expected.html: macOS requires a slightly different
expected output, as it appears to include an unexpected <br> in the paste output that does not appear on iOS or GTK.
* LayoutTests/platform/mac-wk1/fast/events/standalone-image-drag-to-editable-expected.txt: Rebaseline for non-quirks behavior.
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/read-media/pageload-image-expected.txt: Update to
expect PASS.
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/read-media/pageload-video-expected.txt: Ditto.
* Source/WebCore/html/ImageDocument.cpp:
(WebCore::ImageDocument::createDocumentStructure): Use NoQuirksMode.
* Source/WebCore/html/MediaDocument.cpp:
(WebCore::MediaDocument::MediaDocument): Use NoQuirksMode.
* Source/WebCore/html/PluginDocument.cpp:
(WebCore::PluginDocumentParser::createDocumentStructure): Configure view with explicit height and width.
(WebCore::PluginDocument::PluginDocument): Use NoQuirksMode.
* Source/WebCore/loader/SinkDocument.cpp:
(WebCore::SinkDocument::SinkDocument): Use NoQuirksMode.

Canonical link: https://commits.webkit.org/254526@main
